src/core/CMakeLists.txt

```diff
@@ -246,11 +246,15 @@ target_sources(openthread-radio PRIVATE
 if(OT_BUILTIN_MBEDTLS)
```

Comment:
whether we use builtin `mbedtls` in third_party or `mbedtls` provided by application, we need the mbedtls targets to compile. So I think this should be removed?

Reply:
This was added with #4696 to support linking with external mbedTLS. Are you sure we want to revert this?

Reply:
@rlubos Are you using external mbedtls but not using the canonical mbedtls targets name (`mbedtls`, `mbedcrypto` and `mbedx509`)?

Reply:
@wgtdkp, just wanted to make sure you saw the comment from @rlubos in https://github.com/openthread/openthread/pull/4696#issuecomment-600489939.

Reply:
Zephyr specifies it's own CMakeLists.txt for mbedTLS and the library is called `mbedTLS`.

But, as I wrote in the comment linked, if OpenThread prefers to link mbedTLS here unconditionally, I can work it around in Zephyr (or convince Zephyr community to switch to canonical name).

Reply:
I can see that other OT libraries (ncp, cli)  have also gained `mbedtls` target dependency in this commit, so it would likely make sense to give up on this condition for consistency.

Reply:
@rlubos, thanks for your input! I have removed the condition in this PR.

Reply:
@jwhui Thanks for the link, I was careless. @rlubos Now I understand the situation.

I think, leaving openthread linking against nothing doesn't make sense. I believe you need to manually write:
```cmake
target_link_libraries(openthread-ftd PRIVATE mbedTLS)
```
to compile openthread libraries.

I have some options (not necessary be addressed in this PR):
1. Openthread always use canonical mbedTLS target names, and for your case, you can add interface libraries or library ALIAS to adapt the target name `mbedTLS` to one of the caonical mbedtls target names.
2. replace `OT_BUILTIN_MBEDTLS` with `OT_EXTERNAL_MBEDTLS` which gives the external mbedtls library name, and link against `OT_EXTERNAL_MBEDTLS` if it is provided.

thoughts?

Reply:
Thanks for the feedback @wgtdkp, 1 sounds like a viable solution to me, I'll try to add an alias name for the mbedTLS during the next OT update in Zephyr. 

In case I meet some serious blocker, I'll come back to you guys, we might consider the second solution then (which also sounds fine, but that would add some extra complexity one the OpenThread side which might not necessarily be needed).

Reply:
@rlubos Thanks, please don't hesitate to reach out if you have any issue with it.